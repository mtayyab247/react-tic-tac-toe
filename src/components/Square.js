import React from 'react';

const Square = ({row, col, classAttr}) => {
    return(
        <div className={classAttr}>
        </div>
    );
};

export default Square;