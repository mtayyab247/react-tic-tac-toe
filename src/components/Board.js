import React from 'react';
import Square from './Square';
import '../css/board.css';

const Board = () => {
    
    return(
        <div className="board">
            <Square classAttr={"bb br"} row={0} col={0} />
            <Square classAttr={"bb"} row={0} col={1} />
            <Square classAttr={"bb bl"} row={0} col={2} />
            <Square classAttr={"bb br"} row={1} col={0} />
            <Square classAttr={"bb"} row={1} col={1} />
            <Square classAttr={"bb bl"} row={1} col={2} />
            <Square classAttr={"br"} row={2} col={0} />
            <Square classAttr={""} row={2} col={1} />
            <Square classAttr={"bl"} row={2} col={2} />
        </div>
    );
};

export default Board;