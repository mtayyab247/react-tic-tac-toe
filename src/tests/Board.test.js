import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Board from '../components/Board';
import Square from '../components/Square';

Enzyme.configure({
    adapter: new Adapter()
});

describe("<Board /> test", () => {
    it("Should render game board and squares", () => {
        const wrapper = shallow(<Board />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("Should render game square", () => {
        const wrapper = shallow(<Square />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});
