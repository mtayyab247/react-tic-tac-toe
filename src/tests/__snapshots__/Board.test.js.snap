// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> test Should render game board and squares 1`] = `
<div
  className="board"
>
  <Square
    classAttr="bb br"
    col={0}
    row={0}
  />
  <Square
    classAttr="bb"
    col={1}
    row={0}
  />
  <Square
    classAttr="bb bl"
    col={2}
    row={0}
  />
  <Square
    classAttr="bb br"
    col={0}
    row={1}
  />
  <Square
    classAttr="bb"
    col={1}
    row={1}
  />
  <Square
    classAttr="bb bl"
    col={2}
    row={1}
  />
  <Square
    classAttr="br"
    col={0}
    row={2}
  />
  <Square
    classAttr=""
    col={1}
    row={2}
  />
  <Square
    classAttr="bl"
    col={2}
    row={2}
  />
</div>
`;

exports[`<Board /> test Should render game square 1`] = `<div />`;
